libs/client/WebhookClient.lua

```
@@ -0,0 +1,314 @@
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:setChannelId(channelId)
+	self._options.channelId = channelId
+	local bool, err = self:modify({channel_id = channelId})
+
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:setName(name)
+	self._options.name = name
+	local bool, err = self:modify({name = name})
+
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:modify(tbl)
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
+		self._options[k] = v
+	end
+
+	return modify(self, tbl)
+end
+
+function WebhookClient:modifyWithToken(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
+		if k ~= "channelId" then -- You can not change the channelId
+			self._options[k] = v
+		else
+			self:info("modifyWithToken does not accept a channelId key. Use 'modify' to change it.")
+		end
+	end
+	self._api:modifyWebhookWithToken(self._id, self._token, tbl)
+
+	return self
+end
+
+-- Rest of functions
+function WebhookClient:send(content, options) -- return message
+	local file
+	if options then
+		for key, value in pairs(options) do
+			self._options[key] = value -- merge new options
+		end
+
+		if options.file then -- Only one file can be sent at the same time.
+			cntn, err = parseFile(options.file)
+			if err then
+				return nil, err
+			end
+			file = cntn
+		end
+	end
+
+	if file then
+		self._options.file = file
+	end
+
+	local webhookData = self._options
+	local data, err = self._api:executeWebhook(self._id, self._token, { -- We insert these values manually so that we don't send extra (irrevelant) information
+		content = content,
+		username = webhookData.name,
+		avatar_url = webhookData.avatarURL,
+		tts = webhookData.tts,
+		embeds = webhookData.embeds
+	}, webhookData.file)
+
+	if data then
+		return Message(data, self), self
+	else
+		return self:error(err)
```

Should return the error instead of logging it.